### PR TITLE
mark deprecated as deprecated indy_prover_get_credentials_for_proof_req

### PR DESCRIPTION
**Pull request by AxelNennker**
Signed-off-by: Axel Nennker <axel.nennker@telekom.de>

indy_prover_get_credentials_for_proof_req is deprecated - mark it as such

### CODE REVIEW

File: libindy/src/api/anoncreds.rs [hunk 1]
```diff
@@ -1154,6 +1154,7 @@ pub  extern fn indy_prover_close_credentials_search(command_handle: i32,
 /// Annoncreds*
 /// Common*
 /// Wallet*
+#[deprecated(since="1.6.1", note="Please use indy_prover_search_credentials_for_proof_req instead!")]
```
Comment (Artemkaaas) [inline libindy/src/api/anoncreds.rs]: Please add the same for `indy_prover_get_credentials` too

Comment (AxelNennker) [reply libindy/src/api/anoncreds.rs]: please review.

**Comment by jovfer** [comment]
@AxelNennker I think this improvement firstly should be included in branch for patch 1.6.x. So I've changed base branch. Please do not merge this branch with master before this PR will be merged

**Comment by jovfer** [comment]
@AxelNennker @Artemkaaas now we have warnings while building tests. I suggest to put `allow deprecated` into utility and change demo test.
```
Compiling indy v1.6.4 (file:///var/lib/jenkins/workspace/Indy_SDK_CI_PR-1088-NGLTR3W7HVRFLEI7SFQPURZ7AUWCFOBYE7E7J32BHSRQ6MHHOTZA/libindy)
warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/demo.rs:294:9
    |
294 |         indy_prover_get_credentials_for_proof_req(prover_get_credentials_for_proof_req_command_handle,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials': Please use indy_prover_search_credentials instead!
   --> tests/utils/anoncreds.rs:247:19
    |
247 |         let err = indy_prover_get_credentials(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'indy::api::anoncreds::indy_prover_get_credentials_for_proof_req': Please use indy_prover_search_credentials_for_proof_req instead!
   --> tests/utils/anoncreds.rs:307:19
    |
307 |         let err = indy_prover_get_credentials_for_proof_req(command_handle,
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    Finished dev [unoptimized + debuginfo] target(s) in 544.26 secs
```

**Pull request merged**